iOSClient/Networking/NCNetworking.swift

```diff
@@ -163,39 +161,49 @@ import Queuer
         let protectionSpace: URLProtectionSpace = challenge.protectionSpace
         let directoryCertificate = CCUtility.getDirectoryCerificates()!
         let host = challenge.protectionSpace.host
-        let pushNotificationServerProxyHost = URL(string: NCBrandOptions.shared.pushNotificationServerProxy)?.host
-            
+        let certificateSavedPath = directoryCertificate + "/" + host + ".der"
+        var isTrusted: Bool
+
+        defer {
+            #if !EXTENSION
+            DispatchQueue.main.async {
+                if !isTrusted {
+                    (UIApplication.shared.delegate as? AppDelegate)?.trustCertificateError(host: host)
+                }
+            }
+            #endif
+        }
+        
         print("SSL host: \(host)")
         
-        if let serverTrust: SecTrust = protectionSpace.serverTrust {
+        if let serverTrust: SecTrust = protectionSpace.serverTrust, let serverCertificate = SecTrustGetCertificateAtIndex(serverTrust, 0)  {
             
+            // extarct certificate txt
             saveX509Certificate(serverTrust, host: host, directoryCertificate: directoryCertificate)
```

Comment:
You can refactor this a bit to pass the certificate directly instead of the trust. Makes more sense semantically and you can remove the same check from the `saveX509Certificate` function (the one calulating `SecTrustGetCertificateAtIndex(trust, 0)`.

Reply:
oh! true great eye